src/plots/stackedBarPlot.ts

```diff
@@ -141,77 +143,122 @@ export class StackedBar<X, Y> extends Bar<X, Y> {
+     *
+     * @param text
+     * @param bounds
+     * @returns {boolean}
+     */
+    const tryDrawLabel = (text: string, bounds: Bounds, barThickness: number) => {
+      const { x, y } = bounds.topLeft;
+      const width = bounds.bottomRight.x - bounds.topLeft.x;
+      const height = bounds.bottomRight.y - bounds.topLeft.y;
+      const textTooLong = this._isVertical
+        ? ( width > barThickness - (2 * StackedBar._LABEL_PADDING) )
+        : ( height > barThickness - (2 * StackedBar._LABEL_PADDING) );
+
+      if (!textTooLong) {
         const labelContainer = this._labelArea.append("g").attr("transform", `translate(${x}, ${y})`);
         labelContainer.classed("stacked-bar-label", true);
 
-        const writeOptions = {
+        const writeOptions: Typesettable.IWriteOptions = {
           xAlign: "center",
           yAlign: "center",
-        } as Typesettable.IWriteOptions;
-        this._writer.write(text, measurement.width, measurement.height, writeOptions, labelContainer.node());
+        };
+        this._writer.write(text, width, height, writeOptions, labelContainer.node());
       }
 
-      return tooWide;
+      return textTooLong;
     };
 
-    maximumExtents.forEach((maximum) => {
-      if (maximum.extent !== baselineValue) {
-        // only draw sums for values not at the baseline
-
-        const text = this.extremaFormatter()(maximum.extent);
-        const measurement = this._measurer.measure(text);
-
-        const primaryTextMeasurement = this._isVertical ? measurement.width : measurement.height;
-        const secondaryTextMeasurement = this._isVertical ? measurement.height : measurement.width;
-
-        const x = this._isVertical
-          ? primaryScale.scale(maximum.axisValue) - primaryTextMeasurement / 2
-          : secondaryScale.scale(maximum.extent) + StackedBar._STACKED_BAR_LABEL_PADDING;
-        const y = this._isVertical
-          ? secondaryScale.scale(maximum.extent) - secondaryTextMeasurement - StackedBar._STACKED_BAR_LABEL_PADDING
-          : primaryScale.scale(maximum.axisValue) - primaryTextMeasurement / 2;
+    const drawLabelsForExtents = (
+          extents: Utils.Map<string | number, StackExtent<Date | number | string>>,
+          computeLabelTopLeft: (
+              stackEdge: Point,
+              textDimensions: Typesettable.IDimensions,
+              barThickness: number,
+          ) => Point,
+    ) => {
+      extents.forEach((maximum) => {
+        if (maximum.extent !== baselineValue) {
+          // only draw sums for values not at the baseline
+
+          const text = this.extremaFormatter()(maximum.extent);
+          const textDimensions = this._measurer.measure(text);
+
+          const { stackedDatum } = maximum;
+          const { originalDatum, originalIndex, originalDataset } = stackedDatum;
+          const barThickness = Plot._scaledAccessor(this.attr(Bar._BAR_THICKNESS_KEY))(originalDatum, originalIndex, originalDataset);
```

Comment:
does this properly handle the case where we define .barEnd()? in which case each bar will have a potentially different thickness?